### PR TITLE
feat: onboard intercom to transformer proxy

### PR DESCRIPTION
**Pull request by ujjwal-ab**
## Description of the change
Resolves INT-287
- Onboard Intercom to transformer Proxy
- It handles the `408` status code from the Intercom and retries in that case

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/intercom/networkHandler.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+const { proxyRequest, prepareProxyRequest } = require('../../../adapters/network');
+const { processAxiosResponse } = require('../../../adapters/utils/networkUtils');
+const { isHttpStatusSuccess } = require('../../util');
+
+const { RetryableError } = require('../../util/errorTypes');
+
+const errorResponseHandler = (destinationResponse, dest) => {
+  const { status } = destinationResponse;
+  if (status === 408 && !isHttpStatusSuccess(status)) {
```
Comment (sanpj2292) [inline src/v0/destinations/intercom/networkHandler.js:9]:
```suggestion
  if (status === 408) {
```

Comment (ujjwal-ab) [reply src/v0/destinations/intercom/networkHandler.js]: Updated

**Comment by sanpj2292** [comment]
@ujjwal-ab Can we add some proxy test-cases as well in this PR ?

**Comment by ujjwal-ab** [comment]
> @ujjwal-ab Can we add some proxy test-cases as well in this PR ?

Added the test.

File: test/__mocks__/data/intercom/proxy_response.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "https://api.intercom.io/users/test1": {
+    "data": {
+      "type": "error.list",
+      "request_id": "000on04msi4jpk7d3u60",
+      "errors": [{ "code": "not_found", "message": "User Not Found" }]
```
Comment (sanpj2292) [inline test/__mocks__/data/intercom/proxy_response.json:6]: Would the destination send `User Not Found` with `408` status-code ?

Comment (ujjwal-ab) [reply test/__mocks__/data/intercom/proxy_response.json]: Updated

**Review by sanpj2292 — APPROVED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Pull request merged**